### ISSUE TITLE
Documentation/API improvement: SceneReference is serialized so we must check AssetGuidHex not empty, else need IsValid method

### ISSUE THREAD

**Issue by hsandt** [starikcetin/Eflatun.SceneReference#22]
I wanted to check if SceneReference was null before using it, but due to serialization, Unity creates a default instance with invalid fields (sceneAsset is null, strings are empty, other properties cause exception on access).

So, basically, my null check `if (mySceneReference != null)` is always verified, and my code tries to use it even when not set.

Trying to access BuildIndex then causes an obscure error:

> KeyNotFoundException: The given key '' was not present in the dictionary.
System.Collections.Generic.Dictionary`2[TKey,TValue].get_Item (TKey key) (at <7560840559c84ffbb83a08c4696f01f7>:0)
Eflatun.SceneReference.SceneReference.get_Path () (at Library/PackageCache/com.eflatun.scenereference@1.2.2/Runtime/SceneReference.cs:26)
Eflatun.SceneReference.SceneReference.get_BuildIndex () (at Library/PackageCache/com.eflatun.scenereference@1.2.2/Runtime/SceneReference.cs:31)

There's not much to do about it if we want to serialize the whole class, but at least the README documentation should indicate in usage that before using the field, it may be good to check for `if (!string.IsNullOrEmpty(mySceneReference.AssetGuidHex)` or the likes.

I also tried to check `mySceneReference.sceneAsset != null`, but the field is internal so I could not access it. I suppose we could add a `bool IsValid()` method that checks sceneAsset to avoid the more complicated AssetGuidHex check.

**Comment by hsandt** [comment]
In fact, I found what I suggested in a competing (but unmaintained) third-party repository: https://github.com/JohannesMP/unity-scene-reference/blob/master/unity-scene-reference/Assets/source/SceneReference.cs#L39

Check the `IsValidSceneAsset` getter:

```cs
#if UNITY_EDITOR
    // What we use in editor to select the scene
    [SerializeField] private Object sceneAsset;
    private bool IsValidSceneAsset
    {
        get
        {
            if (!sceneAsset) return false;

            return sceneAsset is SceneAsset;
        }
    }
#endif
```

It is even better than my suggestion as it also checks for SceneAsset type, as sceneAsset is merely an object.

**Comment by starikcetin** [comment]
Good idea, I will get to it soon.

**Comment by starikcetin** [comment]
@hsandt Implemented in #24. Will be included in the next release.

**Comment by starikcetin** [comment]
Included in [`1.3.0`](https://github.com/starikcetin/Eflatun.SceneReference/releases/tag/1.3.0).
